pkg/cli/etcdsnapshot/etcd_snapshot.go

```diff
@@ -24,23 +43,18 @@ func Run(app *cli.Context) error {
 }
 
 func run(app *cli.Context, cfg *cmds.Server) error {
-	gspt.SetProcTitle(os.Args[0])
-
-	dataDir, err := server.ResolveDataDir(cfg.DataDir)
-	if err != nil {
+	if err := commandSetup(app); err != nil {
 		return err
 	}
 
-	nodeName := app.String("node-name")
-	if nodeName == "" {
-		h, err := os.Hostname()
-		if err != nil {
-			return err
-		}
-		nodeName = h
+	if len(app.Args()) > 0 {
+		return errors.New("the etcd-snapshot command does not take any arguments")
 	}
 
-	os.Setenv("NODE_NAME", nodeName)
+	dataDir, err := server.ResolveDataDir(cfg.DataDir)
```

Comment:
Maybe return this from commandSetup since it's also a common startup operation?

Reply:
Updated.